src/OpenTelemetry.Contrib.PersistentStorage/PersistentStorageHelper.cs

```diff
@@ -0,0 +1,260 @@
+            var size = CalculateFolderSize(directoryPath);
+            Interlocked.Exchange(ref directorySize, size);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static long GetDirectorySize()
+        {
+            return Interlocked.Read(ref directorySize);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static void WriteAllBytes(string path, byte[] buffer)
+        {
+            File.WriteAllBytes(path, buffer);
+            UpdateDirectorySize(buffer.Length);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static void RemoveFile(string fileName)
+        {
+            var fileInfo = new FileInfo(fileName);
+            var fileSize = fileInfo.Length;
+            fileInfo.Delete();
+            UpdateDirectorySize(fileSize * -1);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static string GetUniqueFileName(string extension)
+        {
+            return string.Format(CultureInfo.InvariantCulture, $"{DateTime.UtcNow:yyyy-MM-ddTHHmmss.fffffffZ}-{Guid.NewGuid().ToString("N", CultureInfo.InvariantCulture)}{extension}");
+        }
+
+        internal static string CreateSubdirectory(string path)
+        {
+            string subdirectoryPath = string.Empty;
+
+            try
+            {
+                string baseDirectory = string.Empty;
+#if !NETSTANDARD
+                baseDirectory = AppDomain.CurrentDomain.BaseDirectory;
+#else
+                baseDirectory = AppContext.BaseDirectory;
+#endif
+
+                string appIdentity = Environment.UserName + "@" + Path.Combine(baseDirectory, Process.GetCurrentProcess().ProcessName);
+                string subdirectoryName = GetSHA256Hash(appIdentity);
+                subdirectoryPath = Path.Combine(path, subdirectoryName);
+                Directory.CreateDirectory(subdirectoryPath);
+            }
+            catch (Exception ex)
+            {
+                StorageEventSource.Log.Error($"Error creating sub-directory {path}.", ex);
+            }
+
+            directorySize = CalculateFolderSize(subdirectoryPath);
+            return subdirectoryPath;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static long UpdateDirectorySize(long fileContentLength)
+        {
+            return Interlocked.Add(ref directorySize, fileContentLength);
+        }
+
+        internal static DateTime GetDateTimeFromBlobName(string filePath)
+        {
+            var fileName = Path.GetFileNameWithoutExtension(filePath);
+            var time = fileName.Substring(0, fileName.LastIndexOf('-'));
+            DateTime.TryParseExact(time, "yyyy-MM-ddTHHmmss.fffffffZ", CultureInfo.InvariantCulture, DateTimeStyles.None, out var dateTime);
+            return dateTime.ToUniversalTime();
```

Comment:
What if `TryParseExact` failed (e.g. the operating system automatically created some file such as `Thumbs.db` or `.fseventsd`)?

Reply:
This needs to be handled accordingly - will fix it.

Reply:
Thanks! Consider addressing this in a separate PR to keep each PR scoped.